### PR TITLE
feat: implement network interfaces list API

### PR DESCRIPTION
**Pull request by Unix4ever**
To be used in the interactive installer to configure networking.

Signed-off-by: Artem Chernyshev <artem.0xD2@gmail.com>

### CODE REVIEW

File: api/machine/machine.proto [hunk 1]
```diff
@@ -776,3 +778,18 @@ message GenerateConfigurationResponse {
...
 }
+
+// NetworkInterface describes a single network interface information.
+message NetworkInterface {
+  repeated string addrs = 1;
+  string name = 2;
+  string mac = 3;
+  int32 mtu = 4;
+  bool up = 5;
+}
```
Comment (andrewrynhard) [inline api/machine/machine.proto:789]: We already have an interface definition in `network.proto`. IMHO we should either a) import that here and add any fields we need to it or 2) have the maintenance service implement the network server (just embed the default unimplemented server) and the implement the interfaces RPC. Given that the pattern we have used so far is to embed the unimplemented server, I would lean towards that.

Comment (Unix4ever) [reply api/machine/machine.proto:789]: oh... It's completely the same there. Yeah that makes perfect sense.

File: internal/app/networkd/pkg/networkd/devices.go [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+		addrs := make([]string, 0, len(ifaceaddrs))
+		for _, addr := range ifaceaddrs {
+			addrs = append(addrs, addr.String())
+		}
+
+		ifmsg := &networkapi.Interface{
+			Index:        uint32(iface.Index),
+			Mtu:          uint32(iface.MTU),
+			Name:         iface.Name,
+			Hardwareaddr: iface.HardwareAddr.String(),
```
Comment (andrewrynhard) [inline internal/app/networkd/pkg/networkd/devices.go:37]: Is this mac address?

**Review by andrewrynhard — APPROVED** [review]

File: internal/app/networkd/pkg/networkd/devices.go (hunk 2 above)
Comment (Unix4ever) [reply internal/app/networkd/pkg/networkd/devices.go:37]: yep

**Pull request merged**